blogposts/2021/release-3.31.md

```diff
@@ -0,0 +1,86 @@
+  - description: "For Docker-Compose and Kubernetes users, the built-in main Postgres and codeintel databases have switched to an alpine Docker image. This requires re-indexing the entire database. This process can take up to a few hours on systems with large datasets."
+    url: https://github.com/sourcegraph/sourcegraph/pull/23697
+    category: Admin
+  - description: "All commit search results for `and`-expressions are now highlighted."
+    url: https://github.com/sourcegraph/sourcegraph/pull/23336
+    category: Search
+  - description: Fixed a result streaming throttling issue that was causing significantly increased latency for some searches. This should improve the time to first rendered results for those searches.
+    url: https://github.com/sourcegraph/sourcegraph/pull/23736
+    category: Search
+  - description: Updating draft merge requests on GitLab from batch changes no longer removes the draft status.
+    url: https://github.com/sourcegraph/sourcegraph/issues/23944
+    category: Batch Changes
+  
+---
+
+Sourcegraph 3.31 is now available! For this release, we introduced:
+
+## Code Insights is now in Private Beta
+
+Code Insights is the first tool ever that lets you to track the historical and future trends of anything in your codebase, from code migrations to API usage to linter overrides – or anything else you can run a Sourcegraph search for. Now in Private Beta, Code Insights can run across thousands of repositories, letting you then filter and explore your insights live. Reach out to your Account Executive for details.
+
+### Code Insights runs over all your repositories
+
+Code insights now runs over all repositories – we've removed the the ~50-70 repo limit.
+
+<div style="text-align:center"><video autoplay loop muted playsinline style="width:625px">
+  <source src="https://sourcegraphstatic.com/blog/3.31/insight_all_repos.mp4" type="video/mp4">
+</video></div>
+
+### Filter Code Insights down to subsets of repositories 
+
+Once your code insights have backfilled your historical data, you can instantly explore and filter to just the repositories you care about using `repo:` and `-repo:` regular expressions, and even [save your filtered views](https://docs.sourcegraph.com/code_insights/explanations/code_insights_filters#filter-persistance-and-sharing). 
+
+<div style="text-align:center"><video autoplay loop muted playsinline style="width:625px">
+  <source src="https://sourcegraphstatic.com/blog/3.31/filter_code_insights.mp4" type="video/mp4">
+</video></div>
+
+## Saved search notifications are migrating to code monitoring
+
+Email notifications for saved searches are being deprecated in favor of code monitoring. There will no longer be an option to enable email notifications for new saved searches, and for any existing saved search with email notifications enabled, there is now a button to automatically migrate to code monitoring. Note that existing email notifications for saved searches will be removed in the future.
```

Comment:
```suggestion
Email notifications for saved searches are being deprecated in favor of code monitoring. Like email notifications for saved searches, code monitoring allows you to keep track of and get notified about changes in your code. Some use cases for code monitors include getting notifications for potential secrets, anti-patterns, or common typos committed to your codebase.

If you are using email notifications for existing saved searches, there is now a button to automatically migrate to code monitoring. Note that existing email notifications for saved searches will be removed in the future.
```

Reply:
No dedicated docs page but the PR documents what users will see nicely (https://github.com/sourcegraph/sourcegraph/pull/23275)